### PR TITLE
Implement .org file rendering in Wiki

### LINKED ISSUE #12832: Feature request: org-mode support in wiki
- Gitea version (or commit ref): 1.12.4
- Git version: 2.16.5
- Operating system: CentOS 7
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

It would be great if I can write Wiki page in org-mode format.
I don't necessarily need fancy WYSIWYG editor on a wiki page. I would be happy if I can clone REPO_NAME.wiki.git and create or edit *.org file locally with emacs, then commit/push.
For now, any *.org files pushed as wiki pages are simply ignored.
I guess it would not be too difficult to make gitea render *.org files as wiki pages, just like org files in repos.

...

### PR DESCRIPTION
**Pull request by LinlyBoi**
Resolves #12832
Gitea supports rendering .org files but Wiki does not support pushing .org files and simply ignores them see the issue mentioned for more details.
Editing: Org file on wiki that was pushed
image
After saving: 
image

The major changes were made to the logic that detects the files pushed to the wiki repo to check for .org files as well as .md files.

If two files with the same name but different extension exist (home.org and home.md for example)  .md will take precedence.

Rendering .org worked initially but there was an error hard coded to throw a warning when the detected markup is not markdown and now it uses the detect markup function for the file extension to add support for both .md and .org

This PR currently supports pushing .org files through the git repo for the wiki.

### CODE REVIEW

**Comment by lunny** [comment]
I think we should introduce a configuration option for the Wiki to choose between markdown and org. This way, we won’t need to handle both formats everywhere, which is not sustainable in the long term.

Additionally, we should add tests to ensure that the new file type works correctly.

**Comment by LinlyBoi** [comment]
Would the configuration be per-repo or instance wide? I think even an option to include both formats is viable since both formats can coexist without issue. I will try to add that but any pointers would be great!

**Comment by lunny** [comment]
> Would the configuration be per-repo or instance wide? I think even an option to include both formats is viable since both formats can coexist without issue. I will try to add that but any pointers would be great!

Per-repository, the configuration name should be `the default format` which will allow both formats. If this option doesn’t exist, any page created through the UI will always default to Markdown.

File: models/repo/repo.go [hunk 1]
```diff
@@ -164,6 +164,7 @@ type Repository struct {
 	OriginalURL         string             `xorm:"VARCHAR(2048)"`
 	DefaultBranch       string
 	DefaultWikiBranch   string
+	DefaultWikiFormat   string
```
Comment (lunny) [inline models/repo/repo.go:167]: Please use `RepoUnit` to store the per repository configuration.

File: modules/setting/repository.go [hunk 2]
```diff
@@ -172,6 +173,7 @@ var (
 		DisableMigrations:                       false,
 		DisableStars:                            false,
 		DefaultBranch:                           "main",
+		DefaultWikiFormat:                       "markdown",
```
Comment (lunny) [inline modules/setting/repository.go:176]: I don't think we need a global configuration?

Comment (LinlyBoi) [reply modules/setting/repository.go:176]: It could be useful to have it for cases where you want every repo globally to support both or either upon creation while still leaving the repo owners the freedom to change it.

File: models/repo/repo.go (hunk 1 above)
Comment (LinlyBoi) [reply models/repo/repo.go:167]: Alright I'll make the changes needed still getting familiar with the codebase apologies.

**Comment by wxiaoguang** [comment]
TBH, I think it needs a complete design instead of adding more patches.

Some edge cases:

* Should it support editing "md" file if "org" is used? How to?
* If both "md" and "org" exist, how to manage? (eg: delete)
* The new design and changed code should be covered by tests.

----
And TBH, actually, the legacy wiki file design is quite problematic, including the "escaping" mechanism, they don't work well.

If let me re-design the wiki, I'd like to make  the path name / wiki title / file name more flexible, and remove all these legacy tricks (avoid unnecessary URL encoding, avoid space and dash hacking, etc).
